README.md

```diff
@@ -34,76 +31,110 @@ notifications:
     recipients:
       # - user@email.com
 env:
-  matrix:
-    - ROS_DISTRO=kinetic  ROS_REPO=ros  TEST=clang-format
-    - ROS_DISTRO=kinetic  ROS_REPO=ros              UPSTREAM_WORKSPACE=https://raw.githubusercontent.com/ros-planning/moveit/kinetic-devel/moveit.rosinstall
-    - ROS_DISTRO=kinetic  ROS_REPO=ros-shadow-fixed UPSTREAM_WORKSPACE=https://raw.githubusercontent.com/ros-planning/moveit/kinetic-devel/moveit.rosinstall
+  global: # default values that are common to all configurations (can be overriden below)
+    - ROS_DISTRO=melodic   # ROS distro to test for
+    - ROS_REPO=ros         # ROS binary repository [ros | ros-shadow-fixed]
+    - TEST_BLACKLIST=      # list packages, for which to skip the unittests
+    - WARNINGS_OK=false    # Don't accept warnings [true | false]
+  matrix:  # define various jobs
+    - TEST=clang-format    # check code formatting for compliance to .clang-format rules
+    - TEST=clang-tidy-fix  # perform static code analysis and compliance check against .clang-tidy rules
+    - TEST=catkin_lint     # perform catkin_lint checks
+    # pull in packages from a local .rosinstall file
+    - UPSTREAM_WORKSPACE=moveit.rosinstall
+    # pull in packages from a remote .rosinstall file and run for a non-default ROS_DISTRO
+    - UPSTREAM_WORKSPACE=https://raw.githubusercontent.com/ros-planning/moveit/$ROS_DISTRO-devel/moveit.rosinstall
+      ROS_DISTRO=kinetic  ROS_REPO=ros-shadow-fixed
```

Comment:
great documentation here!

Reply:
Thanks. 
Obviously your comment in https://github.com/ros-planning/moveit_ci/pull/45/files#r251293176 was referring to the [`.travis.yml` of `moveit_ci`](https://github.com/ros-planning/moveit_ci/blob/88447c8c27b459540ad8275065ec169a6668df60/.travis.yml).